src/app/components/shared/datetime-formats/duration/duration.component.ts

```diff
@@ -19,33 +25,33 @@ export class DurationComponent extends AbstractTemplateComponent<InputType, Dura
 
   // an empty string is used to indicate that the attribute is present
   // eg. <baw-duration humanized> will cause the humanized attribute to be set to an empty string ("")
-  @Input({ transform: booleanAttribute }) public humanized?: boolean;
-  @Input({ transform: booleanAttribute }) public iso8601?: boolean;
-  @Input({ transform: booleanAttribute }) public sexagesimal?: boolean;
+  public humanized = input<boolean, string>(false, { transform: booleanAttribute });
+  public iso8601 = input<boolean, string>(false, { transform: booleanAttribute });
+  public sexagesimal = input<boolean, string>(false, { transform: booleanAttribute });
 
   public update(): void {
-    this.isoDateTime = this.value.toISO();
+    this.isoDateTime = this.value().toISO();
     this.documentText = this.formattedValue();
     this.tooltipText = this.tooltipValue();
   }
 
   // if no format operator is supplied, the component will default to sexagesimal format
   public formattedValue(): string {
-    if (this.humanized) {
-      const relativePrefix = this.value.valueOf() < 0 ? "-" : "";
-      return relativePrefix + toRelative(this.value, { largest: 2, round: true });
-    } else if (this.iso8601) {
-      return this.value.toISO();
-    } else if (this.sexagesimal) {
-      return this.value.toFormat(DurationComponent.DURATION_SEXAGESIMAL);
+    if (this.humanized()) {
+      const relativePrefix = this.value().valueOf() < 0 ? "-" : "";
+      return relativePrefix + toRelative(this.value(), { largest: 2, round: true });
+    } else if (this.iso8601()) {
+      return this.value().toISO();
+    } else if (this.sexagesimal()) {
+      return this.value().toFormat(DurationComponent.DURATION_SEXAGESIMAL);
```

Comment:
read values once

Reply:
For this component, I am able to read the value once in `update` and pass `value` as a parameter to the `formattedValue` and `tooltipValue` methods.

![image](https://github.com/QutEcoacoustics/workbench-client/assets/33742269/c3f776df-e27b-48b8-a165-8029f382ad25)

The alternative is reading `this.value()` three times (in `update`, `formattedValue`, tooltipValue`)